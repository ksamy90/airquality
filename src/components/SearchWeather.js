import React from 'react';
import apijson from '../data.json';

class Weather extends React.Component {
    state = {
        city: 'Delhi',
    }
    onInputChange1 = (e) => {
        const inputData1 = e.target.value;
        this.setState(() => ({ city: inputData1 }));
    };
    onSubmit = (e) => {
        e.preventDefault();
        const city = this.state.city;
        this.props.onsubmit(city);
    };
    render() {
        return (
            <div className="ui segment">
                <h2>Input city to get air quality</h2>
                <p>Access air quality from the following cities:</p>
                <ul>
                    {
                        apijson.map((city) => (
                            <li key={city.name}>{city.name}</li>
                        ))
                    }
                </ul>
                <form onSubmit={this.onSubmit} className="ui form">
                    <div className="field">
                        <label>City</label>
                        <input type="text" name="search1" onChange={this.onInputChange1} value={this.state.city} required />
                        <button className="ui grey basic button">Access Air Quality</button>
                    </div>
                </form>
            </div>
        );
    }
};

export default Weather;
