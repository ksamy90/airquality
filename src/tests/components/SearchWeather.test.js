import React from 'react';
import { shallow } from 'enzyme';
import SearchWeather from '../../components/SearchWeather';

test('should render SearchWeather correctly', () => {
    const wrapper = shallow(<SearchWeather />);
    expect(wrapper).toMatchSnapshot();
});

test('should render SearchWeather correctly with city value', () => {
    const city = 'Delhi'
    const onSubmitSpy = jest.fn();
    const wrapper = shallow(<SearchWeather onsubmit={onSubmitSpy} />);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(onSubmitSpy).toHaveBeenCalledWith(city);
});
