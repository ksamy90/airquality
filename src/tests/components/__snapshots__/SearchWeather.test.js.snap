// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchWeather correctly 1`] = `
<div
  className="ui segment"
>
  <h2>
    Input city to get air quality
  </h2>
  <p>
    Access air quality from the following cities:
  </p>
  <ul>
    <li
      key="Ghaziabad"
    >
      Ghaziabad
    </li>
    <li
      key="Noida"
    >
      Noida
    </li>
    <li
      key="Delhi"
    >
      Delhi
    </li>
    <li
      key="Lucknow"
    >
      Lucknow
    </li>
    <li
      key="Gurugram"
    >
      Gurugram
    </li>
    <li
      key="Faridabad"
    >
      Faridabad
    </li>
    <li
      key="Varanasi"
    >
      Varanasi
    </li>
    <li
      key="Kanpur"
    >
      Kanpur
    </li>
    <li
      key="Patna"
    >
      Patna
    </li>
    <li
      key="Muzaffarpur"
    >
      Muzaffarpur
    </li>
    <li
      key="Mandi Gobindgarh"
    >
      Mandi Gobindgarh
    </li>
    <li
      key="Ludhiana"
    >
      Ludhiana
    </li>
    <li
      key="Jodhpur"
    >
      Jodhpur
    </li>
    <li
      key="Amritsar"
    >
      Amritsar
    </li>
    <li
      key="Ahmedabad"
    >
      Ahmedabad
    </li>
    <li
      key="Panchkula"
    >
      Panchkula
    </li>
    <li
      key="Gaya"
    >
      Gaya
    </li>
    <li
      key="Nagpur"
    >
      Nagpur
    </li>
    <li
      key="Vijaywada"
    >
      Vijaywada
    </li>
    <li
      key="Chandrapur"
    >
      Chandrapur
    </li>
    <li
      key="Chennai"
    >
      Chennai
    </li>
    <li
      key="Bhiwadi"
    >
      Bhiwadi
    </li>
    <li
      key="Mumbai"
    >
      Mumbai
    </li>
    <li
      key="Nashik"
    >
      Nashik
    </li>
    <li
      key="Thane"
    >
      Thane
    </li>
    <li
      key="Aurangabad"
    >
      Aurangabad
    </li>
    <li
      key="Hyderabad"
    >
      Hyderabad
    </li>
    <li
      key="Visakhapatnam"
    >
      Visakhapatnam
    </li>
    <li
      key="Rajamahandevaram"
    >
      Rajamahandevaram
    </li>
    <li
      key="Chittoor"
    >
      Chittoor
    </li>
    <li
      key="Bangalore
    "
    >
      Bangalore
      
    </li>
    <li
      key="Trivandrum"
    >
      Trivandrum
    </li>
  </ul>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label>
        City
      </label>
      <input
        name="search1"
        onChange={[Function]}
        required={true}
        type="text"
        value="Delhi"
      />
      <button
        className="ui grey basic button"
      >
        Access Air Quality
      </button>
    </div>
  </form>
</div>
`;
